### ISSUE TITLE
[tracking] remaining Python 2 packages

### ISSUE THREAD

**Issue by classabbyamp** [void-linux/void-packages#38229]
### core

#38731

- [ ] python
- [ ] python-pip
- [ ] python-tkinter
- [ ] python-setuptools

### buildbot

#38731

<details><summary>waiting on infrastructure migration</summary>

- [ ] buildbot
- [ ] buildbot-slave
- [ ] python-Jinja2
- [ ] python-MarkupSafe
- [ ] python-PyHamcrest
- [ ] python-SQLAlchemy
- [ ] python-Twisted
- [ ] python-attrs
- [ ] python-automat
- [ ] python-cffi
- [ ] python-constantly
- [ ] python-cryptography
- [ ] python-dateutil
- [ ] python-decorator
- [ ] python-enum34
- [ ] python-hyperlink
- [ ] python-idna
- [ ] python-incremental
- [ ] python-ipaddress
- [ ] python-openssl
- [ ] python-pbr
- [ ] python-ply
- [ ] python-pyasn1
- [ ] python-pyasn1-modules
- [ ] python-pycparser
- [ ] python-service_identity
- [ ] python-six
- [ ] python-sqlalchemy-migrate
- [ ] python-sqlparse
- [ ] python-tempita
- [ ] python-zope.interface

</details>

<details><summary>completed</summary>

- [x] GCP-Guest-Environment - #39563
- [x] GConf - #46334
- [x] Pyrex - #39561
- [x] ampache - #38973
- [x] android-studio - #45676
- [x] apache-ant - ab5d182771
- [x] apache-directory-studio-bin - c07689d6
- [x] armagetronad - bcbeca14
- [x] audacity - e86d23fddf8c6c2137e7380a5eb930900effd3f8
- [x] bluefish (build option) - #38607
- [x] boinc - a094163dae2c295f25b92c8760b81e9bdb296b26
- [x] bookworm - #46627
- [x] boost - #38949
- [x] boswars - 909c7f0677
- [x] cbang - #41962
- [x] cegui - #38614
- [x] cegui07 (build option) - #38614 
- [x] chromium - #38434
- [x] chronograf - 98eef89426ad7175e0dd8021deb073daaa3c904e
- [x] claws-mail - #38559
- [x] couchpotato - #39458
- [x] cppcms - bd1420246
- [x] csound - c93ae341acde997b7e302245e440754d8a75f6e0
- [x] dispcalGUI - #44849, #46237
- [x] eclipse - #46515
- [x] ecryptfs-utils-python (subpackage) - #45584
- [x] electron13 - #39921
- [x] electron19 - 4c0a7cacc32e43f6959181bd591b95d4243d19c8
- [x] electron24 - 235934769a5bfb6a33b8049f256eb389d53ae7f4
- [x] escrotum - #40072 
- [x] fah - a68225016c52bf6fc891097a2613ee3871533634
- [x] farstream - #38612
- [x] firefox - #39387
- [x] firefox-esr - #39677
- [x] flightgear - #46515 
- [x] flowcanvas - fa055fe6000356b4272349a88f3ac9e4fd755fb1
- [x] freedroidRPG - #46508
- [x] gcc6 - #46484
- [x] geda - #46509
- [x] getmail - #39459
- [x] ghidra - #49886
- [x] gimp - #43881
- [x] gnubg - 786ed9374ca5538f1f41710a101e928cae511988
- [x] grass - #40347
- [x] gtk+ - 1c83c72da159e9220dd426cb90e73fbdced1614f
- [x] gtkpod - fcb27ef030
- [x] gtksourceview2 - #46515
- [x] julia - #39919
- [x] kaffeine - 51d22db831f63e3983a70726ae48a8b19ccb57d9
- [x] kcachegrind - #46515
- [x] kdesignerplugin - 225545f61588f23749714599eab52d9c5459f105
- [x] kdiagram - b040e799b82272c9c7a572538828ff2f69a37f70
- [x] kinit - 4c8ab251fbe4bfbd321e181b5144ff2c09cb4881
- [x] kio - 5ddaaf473e0725b761af0835f300c6adb7eb9504
- [x] kjsembed - 7057418703b56d59544153715b972470efc14272
- [x] konversation - 262a2cccc48d29358458507e8a77bdab560ac32a
- [x] kronometer - 1774d75aed6bbf04cd0d209ec50249c459bb02b9
- [x] kross - 8b2416061516c6c17d6b47571eaa59001903183f
- [x] ktoblzcheck - #38625
- [x] ladish - #46307
- [x] lensfun - #38950
- [x] libaom - 269ad158539780ca384e9404292548cec94c9055
- [x] libappindicator - #42263
- [x] libdbusmenu-glib - #46515
- [x] libfirm - 80b067a8721e2b377290624661f254f2bdce15e0
- [x] libglade - #43881
- [x] libogdf - 28d583c64c67f1d955ea99a46f767b6accd44e95
- [x] libosmocore - 2d4ab50adc5d6d851a8afc7a0f1cfc28b944174d
- [x] libpqxx - 600757b27056f39d9a533e0199a8c0bad0e915b8
- [x] libproxy-python (subpackage) - #43883
- [x] librsvg - 33d3fc7b0a40f6c8d23831d60b31ff2f6cd1d639
- [x] libselinux - 5d2bc4f4176fa0888239abed6926c60b09549097
- [x] linux4.14 - 8b35257c2c
- [x] liquidwar - #39046
- [x] mc - #38079
- [x] meson - #39071
- [x] mimms - #39460
- [x] minikube - 6d9b047a585ae480ac08bfc36e61d05f99821097
- [x] miro-video-converter - #39461
- [x] moosefs - #44384
- [x] mozjs78 - #46525
- [x] netpbm - 04491c47ebee8a74a1dd166eecca7ed1c88ec8d7
- [x] nitroshare - #46515
- [x] nmap - #44262 
- [x] nutty - #38681
- [x] odroid-u2-kernel - #46493
- [x] offlineimap - #38781
- [x] okteta - 190a085d5482dc841db6e9ef45ad17477732c495
- [x] openbabel - 578a6df46553cc15e8552b208c8c44ddf848af52
- [x] opencv - #38628
- [x] opencv4 - #38628
- [x] opendkim - #46492
- [x] openlierox - #43882
- [x] openmsx - #39564
- [x] pidgin - #46490
- [x] pilot-link - #39462
- [x] pinktrace - #46302
- [x] playonlinux - eeb9324b4ea292b8c94c2d4d94358b003d2d0973
- [x] plex-media-server - #40464
- [x] purple-plugin-pack - #46490 
- [x] pygtk - #43881
- [x] python-Cython - #43881
- [x] python-atomicwrites - #39457
- [x] python-backports - #39457
- [x] python-backports.configparser - #39457
- [x] python-backports.functools_lru_cache - #39457
- [x] python-boto - #39563
- [x] python-cairo - #43881
- [x] python-contextlib2 - #39457
- [x] python-coverage - #39457
- [x] python-dbus - 04451d40f81376bc33c519223cd5ef1f8d78492a
- [x] python-funcsigs - #39457
- [x] python-gobject2 - #43881
- [x] python-hypothesis - #39457
- [x] python-importlib_metadata - #39457
- [x] python-isodate - #38232
- [x] python-mock - #39457
- [x] python-more-itertools - #39457
- [x] python-msp430-tools - #46298
- [x] python-notify - #38232
- [x] python-numpy - #43881
- [x] python-packaging - #39457
- [x] python-parsing - #39457
- [x] python-pathlib2 - #39457
- [x] python-pluggy - #39457
- [x] python-py - #39457
- [x] python-pyserial - #46298
- [x] python-pytest - #39457
- [x] python-rdflib - #38232
- [x] python-rfc6555 - #38781
- [x] python-scandir - #39457
- [x] python-selectors2 - #38781
- [x] python-typing - #39457
- [x] python-urwid - #38435
- [x] python-wcwidth - #39457
- [x] python-zipp - #39457
- [x] qt5 - 871121b5
- [x] qt5-webengine - https://github.com/void-linux/void-packages/pull/49887
- [x] qt5-webkit - c8f0d40dc16
- [x] raul - #39170
- [x] refind - #46297
- [x] resynthesizer - #43881
- [x] scanmem - c30d1852cbe88d3ec8b42c4f1a2f0908e6eb506f
- [x] skanlite - d60613a06f2c8f28332a11a622567b02cf958cde
- [x] skktools - #46515 
- [x] slurm-wlm - c207905a2645fa06c100016884e28eea0f69ca36
- [x] speedometer - #38435<
- [x] spice - #40486
- [x] strongswan - #38890
- [x] sun7i-kernel - #46493
- [x] sydbox - #46302
- [x] syslinux - 142bcd3cd8f9ccbd52240923c66c2c18c9343772
- [x] tacacs - #46515 
- [x] teeworlds - #40476
- [x] telepathy-farstream - #46516
- [x] telepathy-gabble - #46516
- [x] telepathy-glib - 2d42eed19ae18d1f4e0efb8d69158ca99a5c60a4
- [x] telepathy-haze - #46516
- [x] telepathy-idle - #43890
- [x] telepathy-logger - #46516
- [x] telepathy-mission-control - #38638
- [x] telepathy-qt5 - #46516
- [x] telepathy-salut - #46516
- [x] thunderbird - #38919
- [x] trollock - #46512
- [x] tuxpaint - #46542 
- [x] tvheadend - #46515 
- [x] udis86 - 0f0688c82a1d320ab2810b95d142f548cc9cff20
- [x] umbrello - 669cbaf194
- [x] vdrift - #46515 
- [x] webkit2gtk - 6353f833c16d1e817d1f0223dfcf8969cc520634
- [x] wmii - #46510
- [x] wxPython - #46237
- [x] xtrlock - #46512
- [x] yasm - 00d6aedd9b8321443a2c91a98503c0aa991c978e

</details>

**Comment by kruceter** [comment]
# libglade

> Libglade is no longer maintained, as its functionality has been folded into GTK in the form of the [GtkBuilder class](https://developer.gnome.org/gtk3/stable/GtkBuilder.html).

\- https://download.gnome.org/sources/libglade/

libglade does indeed use python 2.

`configure` outputs the following:

```
checking for a Python interpreter with version >= 2.0... python
checking for python... /usr/bin/python
checking for python version... 2.7
```

`libglade-convert.in`'s first lines have the following:

```
#!@PYTHON@
# -*- mode: python -*-

# yes, this requires python 2.x and an XML parser module (eg. PyExpat)
```

I suspect that python 2 is used only for libglade-convert and nothing
else.

## Reverse dependencies

### libglade

- [ ] dropbox (does not use it)

### libglade-devel

- [ ] dropbox (does not use it)

- [ ] libglademm (no reverse dependencies, safe to remove)

- [ ] obconf

- [ ] gfm

- [ ] pygtk (also has a lot of revdeps)

- [ ] ogmrip (upstream states that it *can* be used without gui)

- [ ] darktable ([upstream](https://github.com/darktable-org/darktable/blob/release-4.1.0/README.md) does not mention glade at all).

- [ ] gtk-sharp

- [ ] gtk-sharp2

- [ ] lingot (does not use it)

**Comment by kruceter** [comment]
Please checkmark grass (updated in #40347).

**Comment by icp1994** [comment]
webkit2gtk is completed in https://github.com/void-linux/void-packages/commit/6353f833c16d1e817d1f0223dfcf8969cc520634

**Comment by dkwo** [comment]
nodejs-lts now points to nodejs, which uses python3.

**Comment by classabbyamp** [comment]
@Johnnynator arch seems to have qt5-webengine + python3 patches:
- https://gitlab.archlinux.org/archlinux/packaging/packages/qt5-webengine/-/blob/main/qt5-webengine-python3.patch
- https://gitlab.archlinux.org/archlinux/packaging/packages/qt5-webengine/-/blob/main/qt5-webengine-chromium-python3.patch

**Comment by Johnnynator** [comment]
> @Johnnynator arch seems to have patches for qt5-webengine + python3:
> 
>     * https://gitlab.archlinux.org/archlinux/packaging/packages/qt5-webengine/-/blob/main/qt5-webengine-python3.patch
> 
>     * https://gitlab.archlinux.org/archlinux/packaging/packages/qt5-webengine/-/blob/main/qt5-webengine-chromium-python3.patch


```
Traceback (most recent call last):
  File "/builddir/qt5-webengine-5.15.15/src/core/release/../../3rdparty/chromium/mojo/public/tools/mojom/mojom_parser.py", line 23, in <module>
    from mojom.generate import translate
  File "/builddir/qt5-webengine-5.15.15/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/generate/translate.py", line 17, in <module>
    from mojom.generate import generator
  File "/builddir/qt5-webengine-5.15.15/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/generate/generator.py", line 12, in <module>
    from mojom import fileutil
  File "/builddir/qt5-webengine-5.15.15/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/fileutil.py", line 6, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
```
But not compatible with py 3.12 :)

**Comment by classabbyamp** [comment]
damn, guess that's another point in favour of #46631

**Comment by classabbyamp** [comment]
@Johnnynator can you try the qt5-webengine patches again with `python3.11`?
